src/motioncore/protocols/boolean_gmw/boolean_gmw_gate.cpp

```diff
@@ -764,4 +764,96 @@ const motion::SharePointer MuxGate::GetOutputAsShare() const {
   return result;
 }
 
+ConstantAsBooleanGmwInputGate::ConstantAsBooleanGmwInputGate(std::span<const BitVector<>> input,
+                                                             Backend& backend)
+    : ConstantAsBooleanGmwInputGate::Base(backend),
+      input_(std::vector(input.begin(), input.end())) {
+  input_owner_id_ = 0;
+  bits_ = input_.size() == 0 ? 0 : input_.at(0).GetSize();
+  InitializationHelper();
+}
+
+ConstantAsBooleanGmwInputGate::ConstantAsBooleanGmwInputGate(std::vector<BitVector<>>&& input,
+                                                             Backend& backend)
+    : ConstantAsBooleanGmwInputGate::Base(backend), input_(std::move(input)) {
+  input_owner_id_ = 0;
+  bits_ = input_.size() == 0 ? 0 : input_.at(0).GetSize();
+  InitializationHelper();
+}
+
+void ConstantAsBooleanGmwInputGate::InitializationHelper() {
+  auto& communication_layer = GetCommunicationLayer();
+  auto& _register = GetRegister();
+
+  if (static_cast<std::size_t>(input_owner_id_) >= communication_layer.GetNumberOfParties()) {
+    throw std::runtime_error(fmt::format("Invalid input owner: {} of {}", input_owner_id_,
+                                         communication_layer.GetNumberOfParties()));
+  }
+
+  gate_id_ = _register.NextGateId();
+
+  assert(input_.size() > 0u);           // assert >=1 wire
+  assert(input_.at(0).GetSize() > 0u);  // assert >=1 SIMD bits
+  // assert SIMD lengths of all wires are equal
+  assert(BitVector<>::IsEqualSizeDimensions(input_));
+
+  if constexpr (kVerboseDebug) {
+    GetLogger().LogTrace(
+        fmt::format("Created a ConstantAsBooleanGmwInputGate with global id {}", gate_id_));
+  }
+
+  output_wires_.reserve(input_.size());
+  for (auto& v : input_) {
+    output_wires_.push_back(GetRegister().EmplaceWire<boolean_gmw::Wire>(v, backend_));
+  }
+
+  if constexpr (kDebug) {
+    auto gate_info = fmt::format("gate id {},", gate_id_);
+    GetLogger().LogDebug(fmt::format(
+        "Created a ConstantAsBooleanGmwInputGate with following properties: {}", gate_info));
+  }
+}
+
+void ConstantAsBooleanGmwInputGate::EvaluateSetup() {}
+
+void ConstantAsBooleanGmwInputGate::EvaluateOnline() {
+  GetBaseProvider().WaitSetup();
```

Comment:
this is also not needed anymore

Reply:
Line 820 (GetBaseProvider().WaitSetup();) is already removed.